core/src/main/java/org/apache/cxf/common/spi/ClassGeneratorClassLoader.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    private String getFilePath(String s) {
+        String sep = System.getProperty("file.separator");
+        String relativePath = s.replace('.', sep.charAt(0));
+        String userDir = System.getProperty("user.dir");
+        return userDir + sep + "target" + sep + "dump" + sep + relativePath + ".class";
+    }
+    private void saveClass(String className, byte[] bytes) {
+
+        File file;
+        try {
+            String classFileName = getFilePath(className);
+            String finalFileName = classFileName;
+            file = new File(finalFileName);
+            int i = 1;
+            while (file.exists()) {
+                finalFileName = classFileName.substring(0, classFileName.length() - 6) + String.valueOf(i) + ".class";
+                file = new File(finalFileName);
+                i++;
+            }
+            file.getParentFile().mkdirs();
+            try (FileOutputStream fop = new FileOutputStream(file)) {
+                file.createNewFile();
+                fop.write(bytes);
+                fop.flush();
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    protected Class<?> loadClass(String className, byte[] bytes) {
+        if (DEBUG) {
+            saveClass(className, bytes);
+        }
+        TypeHelperClassLoader loader = getOrCreateLoader();
+        synchronized (loader) {
+            Class<?> cls = loader.lookupDefinedClass(className);
+            if (cls == null) {
+                return loader.defineClass(className, bytes);
+            }
+            return cls;
+        }
+    }
+    protected Class<?> findClass(String className) {
+        return getOrCreateLoader().lookupDefinedClass(className);
+    }
+    private TypeHelperClassLoader getOrCreateLoader() {
+        TypeHelperClassLoader loader = bus.getExtension(TypeHelperClassLoader.class);
+        if (loader == null) {
+            synchronized (bus) {
+                loader = bus.getExtension(TypeHelperClassLoader.class);
+                if (loader == null) {
+                    ClassLoader parent = bus.getExtension(ClassLoader.class);
+                    if (parent == null) {
+                        parent = Thread.currentThread().getContextClassLoader();
+                    }
+                    loader = new TypeHelperClassLoader(parent);
+                    bus.setExtension(loader, TypeHelperClassLoader.class);
+                }
+            }
+        }
+        return loader;
+    }
+
+
+    public static class TypeHelperClassLoader extends ClassLoader {
+        ConcurrentHashMap<String, Class<?>> defined = new ConcurrentHashMap<>();
+
+        TypeHelperClassLoader(ClassLoader parent) {
+            super(parent);
+        }
+        public Class<?> lookupDefinedClass(String name) {
+            return defined.get(name.replace('/', '.'));
```

Comment:
It seems like we have recurrent patterns: 
 - `name.replace('/', '.')` slashes to periods (used many times)
 - `ASMHelper:periodToSlashes` periods to slashes
We could extract those to `org.apache.cxf.common.util.StringUtils` (and actually drop one method from `ASMHelper` interface)